### DISCUSSION TITLE
MFC visual studio solution porting to xmake

### DISCUSSION THREAD

**Discussion by davvid3d** [xmake-io/xmake#5203] [Q&A]
Hello,

I am trying to port a visual studio 2017 MFC solution to xmake build only, I am struggling to make MSXML3 compile, to thl file is generated correctly but not included in the compilation making it impossible to succeed because of the pure virtual classes.

Have you already faced such issue ? do you have any advice ?

Thanks in advance,

**Comment by waruqi** [comment]
I haven't encountered anything like that, if you're building an mfc application you could try using `add_rules(‘win.sdk.mfc.shared_app’)`, but I haven't tested that lately either.

https://github.com/xmake-io/xmake/blob/master/xmake/rules/winsdk/mfc/xmake.lua
